cmake/DLAF_AddTest.cmake

```diff
@@ -141,11 +141,25 @@ function(DLAF_addTest test_target_name)
 
     # APPLE platform does not support thread binding
     if (NOT APPLE)
-      list(APPEND _TEST_ARGUMENTS "--pika:use-process-mask")
+      # TODO pika:bind=none?
+      list(FILTER _PIKA_EXTRA_ARGS_LIST EXCLUDE REGEX "--pika:use-process-mask")
```

Comment:
check this with @msimberg

Reply:
I'd leave this as is, i.e. not use `pika:bind=none`. Pasting my summary from slack:

> No, they're largely orthogonal, though I admit the `bind=none` case and macos is in practice not well tested and may do surprising things.

> `--pika:bind` controls how _individual_ threads should be bound to hardware, if at all. They can be bound to a numa domain, core, pu, or not at all.

> `--pika:(ignore/use)-process-mask` controls what resources are available to start with to the whole runtime.

> In theory I think `--pika:bind=none --pika:use-process-mask` would choose the number of threads for the runtime (assuming `--pika:threads` isn't also set) based on the process mask, but then not bind the threads to any particular thread. In practice I'm not sure it works exactly like that because it's badly tested (and an unusual case).

About the actual change here, is the idea to move `--pika:use-process-mask` last in the list? If yes, why?

Reply:
> About the actual change here, is the idea to move `--pika:use-process-mask` last in the list? If yes, why?

It is just a leftover because I copy-pasted the same mechanism for all options. The idea is that I remove the option in case the user already specified it, in order to overwrite the value with the one chosen by this logic. In this case there is no need, since the option does not have a value and it will result, as you said, in just moving it at the end of the list.